packages/wallet/core/src/state/local/memory.ts

```diff
@@ -38,119 +41,123 @@ export class MemoryStore implements Store {
     return out as T
   }
 
-  private getSignatureKey(signer: Address.Address, subdigest: Hex.Hex): string {
-    return `${signer.toLowerCase()}-${subdigest.toLowerCase()}`
+  private getSignatureKey(signer: Address.Checksummed, subdigest: Hex.Hex): `${Address.Checksummed}-${Hex.Hex}` {
+    return `${signer}-${subdigest}`
   }
 
-  private getSapientSignatureKey(signer: Address.Address, subdigest: Hex.Hex, imageHash: Hex.Hex): string {
-    return `${signer.toLowerCase()}-${imageHash.toLowerCase()}-${subdigest.toLowerCase()}`
+  private getSapientSignatureKey(
+    signer: Address.Checksummed,
+    subdigest: Hex.Hex,
+    imageHash: Hex.Hex,
+  ): `${Address.Checksummed}-${Hex.Hex}-${Hex.Hex}` {
+    return `${signer}-${imageHash}-${subdigest}`
   }
 
   async loadConfig(imageHash: Hex.Hex): Promise<Config.Config | undefined> {
-    const config = this.configs.get(imageHash.toLowerCase() as `0x${string}`)
+    const config = this.configs.get(imageHash)
     return config ? this.deepCopy(config) : undefined
   }
 
   async saveConfig(imageHash: Hex.Hex, config: Config.Config): Promise<void> {
-    this.configs.set(imageHash.toLowerCase() as `0x${string}`, this.deepCopy(config))
```

Comment:
This shouldn't have been removed because its a Hex value not a checksummed address